server/channels/store/sqlstore/channel_store.go

```diff
@@ -2020,6 +2020,83 @@ func (s SqlChannelStore) GetChannelMembersTimezones(channelId string) ([]model.S
 	return dbMembersTimezone, nil
 }
 
+type channelsWithUnreadsAndWithMentionsQueryResult struct {
+	Id            string
+	Type          string
+	TotalMsgCount int
+	LastPostAt    int64
+	MsgCount      int
+	MentionCount  int
+	NotifyProps   model.StringMap
+	LastViewedAt  int64
+}
+
+func (s SqlChannelStore) GetChannelsWithUnreadsAndWithMentions(ctx context.Context, channelIDs []string, userID string, userNotifyProps model.StringMap) ([]string, []string, map[string]int64, error) {
```

Comment:
A lot of the logic has now moved from the app layer to the store layer. This method needs a lot of unit tests. Needs to have atleast 3 cases for the 3 types of notify props, and also should test the `hasUnreads` logic.